testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -1584,6 +1585,69 @@ def test_replication_with_10_attachments(setup_client_syncgateway_test):
     # Verify docs show up in sync_gateway's changes feed
     client.verify_docs_in_changes(url=sg_url, db=sg_db, expected_docs=cbl_docs, auth=session)
 
+@pytest.mark.listener
+@pytest.mark.syncgateway
+@pytest.mark.attachments
+def test_replication_with_10_attachments_forsequence(setup_client_syncgateway_test):
+    """
+    1.  Start LiteServ and Sync Gateway
+    2.  Create database
+    3.  Create 10 docs with 10 attachments with 2MB each for attachment.
+    4.  Start continuous push replication from ls_db to sg_db
+    5.  Verify replication is completed and all docs pushed to sg.
+    """
+
+    log_info("Running 'test_replication_with_10_attachments' ...")
+
+    cluster_config = setup_client_syncgateway_test["cluster_config"]
+    sg_mode = setup_client_syncgateway_test["sg_mode"]
+    sg_url = setup_client_syncgateway_test["sg_url"]
+    sg_url_admin = setup_client_syncgateway_test["sg_admin_url"]
+
+    ls_url = setup_client_syncgateway_test["ls_url"]
+
+    log_info("ls_url: {}".format(ls_url))
+    log_info("sg_url: {}".format(sg_url))
+    log_info("sg_url_admin: {}".format(sg_url_admin))
+
+    ls_db = "ls_db"
+    sg_db = "db"
+    num_docs = 25
+    channels = ["ABC"]
+
+    # Reset cluster to ensure no data in system
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    client = MobileRestClient()
+    client.create_database(ls_url, ls_db)
+    client.create_user(sg_url_admin, sg_db, "autotest", password="password", channels=channels)
+    session = client.create_session(sg_url_admin, sg_db, "autotest")
+
+    # add dosc to mobile rest client
+    abc_docs = document.create_docs(doc_id_prefix="abc_docs", number=num_docs, channels=channels)
+
+    user_one_docs = client.add_bulk_docs(url=sg_url, db=sg_db, docs=abc_docs, auth=session)
+    assert len(user_one_docs) == num_docs
+
+    # doc with 2.36 PNG attachment
+    sg_docs = add_2MB_docs(url=sg_url, db=sg_db, number=num_docs, id_prefix="2MB_attachment_k", auth=session, channels=channels, attachment_name="golden_gate_large.jpg", num_of_attachments=20)
+    assert len(sg_docs) == num_docs
+
+    # Start continuous push replication from ls_db -> sg_db
+    repl_id = client.start_replication(
+        url=ls_url,
+        continuous=True,
+        from_url=sg_url_admin, from_db=sg_db,
+        to_db=ls_db
+    )
+    time.sleep(180)
```

Comment:
why this sleep needed if we are anyways waiting for idle status?

Reply:
do not remember why I have added. As status check is there whether it is idle or not, I removed sleep.